### PR TITLE
Add support for /projects/{project_id}/revisions/{commit_hash}/tasks

### PR DESCRIPTION
**Pull request by savannaendicott**
https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#list-tasks-by-project-and-commit

### CODE REVIEW

File: src/evergreen/api.py [hunk 1]
```diff
@@ -544,12 +538,24 @@ def tasks_by_project(self, project_id: str, statuses: Optional[List[str]] = None
...
         """
-        url = self._create_url(
-            "/projects/{project_id}/versions/tasks".format(project_id=project_id)
-        )
+        url = self._create_url(f"/projects/{project_id}/versions/tasks")
         params = {"status": statuses} if statuses else None
         return [Task(json, self) for json in self._paginate(url, params)]  # type: ignore[arg-type]
 
+    def tasks_by_project_and_commit(
+        self, project_id: str, commit_hash: str, params: Dict = None
```
Comment (dbradf) [inline src/evergreen/api.py:546]: [nit] Since `params` can be `None`, I think it should be an `Optional[Dict]`

**Review by dbradf — APPROVED** [review]

File: src/evergreen/api.py (hunk 1 above)
Comment (savannaendicott) [reply src/evergreen/api.py]: nice catch, thanks!

**Comment by savannaendicott** [comment]
Ready to merge, thank you!

**Comment by dbradf** [comment]
evergreen merge

**Pull request merged**